web/docs/internal/keystroke-lifecycle.md

```diff
@@ -0,0 +1,301 @@
+# The Keyman Engine for Web keystroke lifecycle
+
+The phases of Keyman Engine for Web keystroke processings are as follows:
+
+1. Keystroke pre-processing: event interception + interpretation (pre-processing)
+2. Keystroke processing: keystroke main-group rule evaluation
+3. Keystroke post-processing: keystroke post-keystroke evaluation + side-effects
+4. Keystroke resolution: UI state updating
+
+Keyman Engine for Web currently has two different builds, each used in specific
+contexts:
+
+* app/browser - for direct use on a page or site within a web browser
+* app/webview - for integration with a host app, such as Keyman itself
+
+The overall phases for the keystroke lifecycle are the same for both; the
+implementation is merely a bit different for each within specific phases.  These
+differences are documented after the common structure and patterns for each
+phase.
+
+Also note that Keyman Engine for Web supports keystroke processing for different
+keyboard specification formats:
+- JS
+- KMX (via web-core)
+- KMX+ / LDML (via web-core)
+
+The `KeyboardProcessor` classes each handle keystroke processing for one
+specification format while adhering to a common interface exposed and accessible
+to the other layers of the keystroke lifecycle.
+
+```mermaid
```

Comment:
I strongly suggest using View File in order to visibly see the resulting diagram.

As of the current commit, that'd be https://github.com/keymanapp/keyman/blob/dd77c268fc38dbf86ccf9194604fcb4f611e863e/web/docs/internal/keystroke-lifecycle.md